### PR TITLE
Update v1/ yaml to be swagger 2.0 compliant

### PR DESCRIPTION
**Pull request by clarakosi**
Modified the optional parameters with a set of paths with required parameters
to make code swagger 2.0 complaint.

Also, deleted trending.yaml & .js since it has been deprecated.

Bug: T218218

### CODE REVIEW

**Comment by Pchelolo** [comment]
The PR LGTM. 

As for the errors you seem to have submitted the spec built from the test config. Instead we need to validate the example config. See https://github.com/wikimedia/restbase/blob/master/test/features/router/buildTree.js#L1 - there we are building and getting the spec from the example config.

Let's try to rerun the validator with the proper spec and see what happens. Some of the error messages do not seem to be correct and might be errors in the validator. We might wanna use some other swagger validator implementation. There's a bunch, for example 'swagger-tools' npm package, but we want to find the one that supports both swagger 2 and swagger 3.

**Comment by Pchelolo** [comment]
I've checked some tools that exist for swagger validation and I think the swagger-cli npm package provides the most meaningful set of errors. It is also capable of validating openapi 3.0, so I believe that's what we should use

**Comment by Pchelolo** [comment]
I've tested myself and the package only gives you one error per run (sucks). Here's a pointer on how to fix the first error: 

https://github.com/wikimedia/hyperswitch/blob/master/lib/router.js#L513

**Comment by clarakosi** [comment]
Ok. I'll look into it this morning

**Comment by clarakosi** [comment]
The remaining errors are in regard to `patternProperties` which is not supported and `oneOf` which is part of 3.0.

Still need to: create unit-tests + create PR for Hyperswitch

**Comment by Pchelolo** [comment]
> The remaining errors are in regard to patternProperties which is not supported and oneOf which is part of 3.0.

Both are used only in the section editing API which is being deprecated, so feel free to remove those. Otherwise LGTM.

File: test/features/schema_tests.js [hunk 1]
```diff
@@ -101,3 +102,30 @@ parallel('Responses should conform to the provided JSON schema of the response',
...
+                return preq.post({uri: `http://online.swagger.io/validator/debug`, body: swagger})
+                    .then((res) => {
+                        // if valid will return an empty object: {}
+                        if (Object.keys(res.body).length !== 0) {
+                            throw new Error(`Swagger specification is not valid: ${res.body}`)
+                        }
+                    })
+            })
+    });
+});
```
Comment (Pchelolo) [inline test/features/schema_tests.js]: This has to be rebased after #1101 is merged and the spec validity test has to be moved here https://github.com/wikimedia/restbase/pull/1101/files#diff-ff16a5ee31f1814844151cee8c98d8aaR38

File: test/features/schema_tests.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 // mocha defines to avoid JSHint breakage
 /* global it, before */
 
+const ServiceRunner = require('service-runner');
```
Comment (Pchelolo) [inline test/features/schema_tests.js]: Why?

File: test/features/schema_tests.js [hunk 3]
```diff
@@ -34,9 +35,16 @@ parallel('Responses should conform to the provided JSON schema of the response',
...
         return preq.get({ uri: `${server.config.baseURL()}/?spec` })
-        .then((res) => {
-            // TODO: validate the openapi spec!
-        });
+            .then((res) =>  {
+                return preq.post({uri: `http://online.swagger.io/validator/debug`,
+                    body: JSON.stringify(res.body)})
+                    .then((res) => {
+                        // if valid will return an empty object: {}
+                        if (Object.keys(res.body).length !== 0) {
```
Comment (Pchelolo) [inline test/features/schema_tests.js]: Event better: assert.deepEquals({}, res.body, 'Spec must have no validation errors')

File: test/features/schema_tests.js [hunk 4]
```diff
@@ -34,9 +35,16 @@ parallel('Responses should conform to the provided JSON schema of the response',
 
     it('should expose valid OpenAPI spec', () => {
         return preq.get({ uri: `${server.config.baseURL()}/?spec` })
-        .then((res) => {
-            // TODO: validate the openapi spec!
-        });
+            .then((res) =>  {
+                return preq.post({uri: `http://online.swagger.io/validator/debug`,
```
Comment (Pchelolo) [inline test/features/schema_tests.js]: I think it's better to install a dev dependency to swagger-cli and use it to validate. Lett dependencies on internet and external service.

Comment (clarakosi) [reply test/features/schema_tests.js]: One of my concerns with swagger-cli is that it uses swagger-parser which is using a deprecated dependency: https://github.com/APIDevTools/swagger-parser/pull/109

File: test/features/schema_tests.js (hunk 2 above)
Comment (clarakosi) [reply test/features/schema_tests.js]: From the previous version. Will remove.

File: test/features/schema_tests.js (hunk 4 above)
Comment (Pchelolo) [reply test/features/schema_tests.js]: hm. ok. Let's use the online one for now then and file a ticket to keep an eye on the swagger-cli thing.

**Comment by Pchelolo** [comment]
Waiting for travis and merging.

File: v1/graphoid.yaml [hunk 5]
```diff
@@ -34,7 +34,7 @@ paths:
         '200':
           description: The PNG render of the requested graph.
           schema:
-            type: blob
+            type: string
```
Comment (d00rman) [inline v1/graphoid.yaml]: PNGs are binary blobs, not strings, so we need `type: file` here (mental note: we'll need to change this to be v3-compliant)

**Comment by Pchelolo** [comment]
Once @d00rman comment is fixed, let's not wait for travis. I've run tests locally to and it's all good. Travis is on vacation.

File: v1/graphoid.yaml (hunk 5 above)
Comment (clarakosi) [reply v1/graphoid.yaml]: Done

**Pull request merged**